src/apps/testapps/testH3Memory.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file
+ * @brief tests main H3 core library memory management.
+ *
+ *  usage: `testH3Memory`
+ */
+
+#include <math.h>
+#include "alloc.h"
+#include "geoCoord.h"
+#include "h3api.h"
+#include "test.h"
+#include "utility.h"
+
+#ifndef H3_ALLOC_PREFIX
+#error "This test can only be run with a prefix for alloc functions"
+#endif
+
+static int failAlloc = 0;
+
+void* H3_MEMORY(malloc)(size_t size) {
```

Comment:
I think the test would be more explicit if you actually called these `custom_malloc` or what have you, then failed the test if `H3_ALLOC_PREFIX != "custom_"`

Reply:
This turns out to be simpler since a separate copy of the library is needed (so that both code paths can be tested)